source/extensions/common/wasm/context.cc

```diff
@@ -858,11 +858,7 @@ BufferInterface* Context::getBuffer(WasmBufferType type) {
 void Context::onDownstreamConnectionClose(CloseType close_type) {
   ContextBase::onDownstreamConnectionClose(close_type);
   downstream_closed_ = true;
-  // Call close on TCP connection, if upstream connection closed or there was a failure seen in
-  // this connection.
-  if (upstream_closed_ || getRequestStreamInfo()->hasAnyResponseFlag()) {
-    onCloseTCP();
-  }
```

Comment:
Quick thought, should we add:
```
if (!upstream_closed_ && !getRequestStreamInfo()->hasAnyResponseFlag()) {
  onUpstreamConnectionClose(CloseType::Unknown);
}
```
to catch connection failures and always emit the upstream connection close event to Wasm plugins?

Reply:
It would be called before `onCloseTCP`, so state wouldn't be destructed yet. But sure, we can leave it as-is if you prefer.